src/ItemList.js

```diff
@@ -1,15 +1,65 @@
-import React from 'react';
-
+import React, { useState, useEffect } from 'react';
 import Item from './Item';
 import { useHistory } from 'react-router-dom';
 
+// [filter-list] 5. Add function for 'X' button to clear query value when clicked
 
 function ItemList(props) {
+  const [query, setQuery] = useState('');
+  const [queryArray, setQueryArray] = useState([]);
   let history = useHistory();
   const redirect = () => {
     history.push('/additems');
   };
 
+  useEffect(() => {
+    // [filter-list] Remove accidental space character from query and make lowercase
+    const transformQuery = () => {
+      let transformedQuery = query.toLowerCase().trim();
+      const regexSpecialChars = [
+        '[',
+        '\\',
+        '^',
+        '$',
+        '.',
+        '|',
+        '?',
+        '*',
+        '+',
+        '(',
+        ')',
+      ];
+
+      if (
+        regexSpecialChars.includes(
+          transformedQuery[transformedQuery.length - 1],
+        )
+      ) {
+        transformedQuery = transformedQuery.slice(
+          0,
+          transformedQuery.length - 1,
+        );
+      }
+      return transformedQuery;
+    };
+
+    const filterRegex = new RegExp(transformQuery());
+
+    // [filter-list] 2. Comparison function to filter shopping list and create a search results array
+    const resultsArray = props.list.filter((itemObj) => {
+      return filterRegex.test(itemObj['itemName'].toLowerCase());
+    });
```

Comment:
What about something like: 
                    `itemObj.itemName.toLowerCase().replace(/[\W_]/g, '').includes(query.toLowerCase()),`

instead of all of the above. I'm still getting an error if I enter 2 backslashes.

Reply:
@laurenmbeatty I was trying to account for a case where the user might actually want to search for one of the "forbidden" characters, by only allowing them if they weren't the last character in the string. But you're right, if more than one character at the end of the string is a JS special character, the function will only omit the last one. I'm comfortable with implementing this solution if we want to just prevent the user from using those characters all together!

Reply:
Oh, I see. Totally up to you all.

Reply:
I went ahead and implemented it, and it did the trick! If anyone else thinks we should allow users to search for non-word characters, let me know and we can try to brainstorm an alternate solution.

Reply:
If we want to keep special characters in the search query, we could just escape them in the regex, i.e. add a `\` before each of them to let the regex know they should be read as literals. This is how MDN [suggests](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping) to do this:
`string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string`

Reply:
Oh that's so smart, Daria! I'll update the code with this change!

Reply:
Nice!